### ISSUE TITLE
Delete Meta Data Without ICC Color Profile

### ISSUE THREAD

**Issue by ykamei** [ImageOptim/ImageOptim#163]
Thank you for your very useful application.

May I request you the new feature?
I want to delete any metadata without ICC Color Profile.
But when I select the option "Strip JPEG metadata", it delete color profile too.
I'm very happy if you support the option to preserve color profile.

**Comment by pathawks** [comment]
As Color Profiles are being supported in more places (especially on Apple devices) it _would_ be nice to keep color profiles while stripping everything else:+1:

**Comment by curby** [comment]
Related to, and possibly a dupe of, #96

**Comment by ykamei** [comment]
curby, Thanks for your replying.
Yes, that issue is similar to this.
But my suggestion is to preserve ICC profile except EXIF and GPS info.
Even if we use sRGB, I want to show images correctly on the display which has wide color range(like Adobe RGB).

**Comment by curby** [comment]
Yes, the same suggestion came out of #96; namely, to preserve color profile metadata while stripping other metadata, yielding output images that do not visibly differ from their inputs.  I think we're all on the same page here.

**Comment by ykamei** [comment]
sorry, I seem to have misunderstood about #96 .
I will write something about this issue if I need.
thanks!

**Comment by mustii82** [comment]
(Y)

**Comment by roryokane** [comment]
The ImageOptim website has a page about [why the software strips embedded color profiles](https://imageoptim.com/color-profiles.html). The most relevant part:

> In practice:
>
> - The majority of typical consumer monitors have a color profile similar to the standard sRGB profile, and can't display any “better” profiles.
> - Almost all images, especially on the Web, were designed to be displayed in the sRGB color profile. When displayed correctly, they look the same on a high-end “wide gamut” monitor as on an average sRGB monitor.
> - Not all browsers and image viewers [support color profiles correctly](https://kornel.ski/en/color). While basic support has improved in recent years, there are still cases where it's buggy or misconfigured, so it can't be relied upon for anything fancy.
>
> As a result, for most users, embedded color profiles provide no value. Even for the users of high-end monitors there's rarely any benefit, because most images don't contain extremely saturated colors that require a special profile.
>
> However, the use of embedded color profiles has a substantial cost in file size — the profiles can balloon an image by over 100KB! For small images that may multiply their file size.

I think that any argument for adding a setting to preserve embedded color profiles should respond to those points.

---

I think the first point above, that typical monitors are sRGB, is no longer a good argument to omit this feature. A significant proportion of image viewing happens on iPhones, and iPhone screens support the P3 color profile. The screens of Apple iPads, MacBooks, and iMacs also support P3, and I think the percentage of image-viewing done on those devices as a group is also significant.

However, I do not have the knowledge or data to refute the second point’s claim that almost all images with stripped color profiles will look the same as with preserved color profiles, even on P3-compatible monitors.
